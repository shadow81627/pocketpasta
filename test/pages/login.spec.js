import Page from '@/pages/login.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import BootstrapVuePlugin from 'bootstrap-vue';
import Vue from 'vue';

const localVue = createLocalVue();
let vuetify;

Vue.use(Vuetify);
localVue.use(BootstrapVuePlugin);

const factory = () =>
  shallowMount(Page, {
    localVue,
    vuetify,
    propsData: {
      label: 'Job Title',
      value: 'Full Stack Web Developer',
    },
  });

describe('login', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
