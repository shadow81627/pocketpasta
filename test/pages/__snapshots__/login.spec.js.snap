// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders properly 1`] = `
"<div class=\\"container\\" label=\\"Job Title\\" value=\\"Full Stack Web Developer\\">
  <div>
    <h1>Login</h1> <span><auth-stub></auth-stub></span>
  </div>
</div>"
`;
